crates/topos/src/config/genesis/mod.rs

```diff
@@ -44,4 +44,45 @@ impl Genesis {
             None => Vec::default(),
         }
     }
+
+    /// Parse the validators from the `extraData` field of the genesis file.
+    /// The `extraData` is patted with 32 bytes, and the validators are RLP encoded.
+    /// Each validator is 20 bytes, with a SEAL at the end of the whole list (8 bytes)
+    #[allow(dead_code)]
+    pub fn validators(&self) -> Vec<String> {
```

Comment:
@gruberb After diving a bit deeper into how the `extraData` is encoded I came across several details.
I think you have the correct idea of removing the `vanity` from the `extraData` and decoding the remainder with RLP, but after that, the RLP-decoded data is not being handled properly.
The RLP-decoded data should be retrieved as an RLP "object" (I don't know how that works in Rust). From the object, you should be able to infer the validator addresses according to the length of the addresses which is 20 bytes (40 characters) indeed.

I made a dirty script in JavaScript to decode the same `genesis.json` file as in the tests and got the 4 validator public addresses as expected. Posting the script here for reference:
```
const fs = require('fs');
const rlp = require('ethereumjs-util').rlp;

// Function to decode extraData
function decodeExtraData(extraData) {
    const ADDRESS_LENGTH = 20; // bytes
    const VANITY_LENGTH = 64; // characters (32 bytes)

    // Remove the '0x' prefix
    const data = extraData.slice(2);

    // Remove the vanity data
    const validatorsData = data.slice(VANITY_LENGTH, data.length);
    
    const validators = [];
    const rlpDecoded = rlp.decode(`0x${validatorsData}`);
    rlpDecoded[0].forEach((rlpItem) => {
        const potentialAddress = rlpItem[0];
        if (potentialAddress.length === ADDRESS_LENGTH) {
            validators.push(`0x${potentialAddress.toString('hex')}`);
        }
    });
    return validators;
}

// Read the genesis file
fs.readFile(process.argv[2], 'utf8', (err, data) => {
    if (err) {
        console.error('Error reading file:', err);
        return;
    }

    // Parse the JSON data
    const genesis = JSON.parse(data);

    // Decode the extraData field
    const extraData = genesis.genesis.extraData;
    const validators = decodeExtraData(extraData);
    console.log('Validators:', validators);
});
```

Output:
```
Validators: [
  '0x100d617e4392c02b31bdce650b26b6c0c3e04f95',
  '0x92183cff18a1328e7d791d607589a15d9eee4bc4',
  '0xb4973cdb10894d1d1547673bd758589034c2bba5',
  '0xc16d83893cb61872206d4e271b813015d3242d94'
]
```

You can check the validator addresses from the secrets as follows:
```
$ ./polygon-edge secrets output --data-dir ./test-chain-1


[SECRETS OUTPUT]
Public key (address) = 0x100D617E4392C02B31bdCe650b26b6c0c3E04F95
BLS Public key       = 0xae7711044926a23c1462754cbb0d1b43fb91fc8fd18bf5c81edb4de15124203157657bf7a8e86d9a3be5f32de725f3c4
Node ID              = 16Uiu2HAkxKTnwPL3eZmFkiKYJiG3um9uraPh21XhsJEPNm8juhy3

$ ./polygon-edge secrets output --data-dir ./test-chain-2


[SECRETS OUTPUT]
Public key (address) = 0x92183Cff18A1328E7d791D607589A15d9EeE4bC4
BLS Public key       = 0xb45118f9e430d94f424019bb8702e004db5dad5725ab1a5346b0aaad556935189c47df5e401988527ce880bb1e2492ce
Node ID              = 16Uiu2HAmQfaE4bjJMVCwzigAUgp9eLcGQz8HZpURqEPNjtfDwge8

$ ./polygon-edge secrets output --data-dir ./test-chain-3


[SECRETS OUTPUT]
Public key (address) = 0xB4973Cdb10894D1D1547673bD758589034C2BBa5
BLS Public key       = 0xb9833912ee2eab270a1204f3f9e58c5f2be603cc2ce32f5467e2a8246bb6b25a7908b39a8a0ed629a689da376b5cdd2d
Node ID              = 16Uiu2HAm5tS7AdBhhtQ2JuTtyy2U4uFqsiW57vof5fiyHPRpwboD

$ ./polygon-edge secrets output --data-dir ./test-chain-4


[SECRETS OUTPUT]
Public key (address) = 0xC16d83893cB61872206D4e271B813015D3242d94
BLS Public key       = 0xa468068169523df684362de6a5b729c8db400958bfd4f6d4e3646cc640f3d241253a21b00f05ff97545f535b36b31c7b
Node ID              = 16Uiu2HAmFjGqEUYSnKkoqURRu8bZaWpuGdxBBqW15KFs4BsxFxdp
```